contracts/src/TokenRemote/NativeTokenRemote.sol

```diff
@@ -1,395 +1,26 @@
-    function _getNativeTokenRemoteStorage()
-        private
-        pure
-        returns (NativeTokenRemoteStorage storage $)
-    {
-        // solhint-disable-next-line no-inline-assembly
-        assembly {
-            $.slot := _NATIVE_TOKEN_REMOTE_STORAGE_LOCATION
-        }
-    }
-
-    /**
-     * @notice The address where the burned transaction fees are credited.
-     *
-     * @dev Defined as BLACKHOLE_ADDRESS at
-     * https://github.com/ava-labs/subnet-evm/blob/v0.6.0/constants/constants.go
-     * C-Chain value found at https://github.com/ava-labs/coreth/blob/v0.13.2/constants/constants.go
-     * It is a system-reserved address by default in subnet-evm and coreth, and transfers cannot be sent here manually.
-     */
-    address public constant BURNED_TX_FEES_ADDRESS = 0x0100000000000000000000000000000000000000;
-
-    /**
-     * @notice The address where native tokens are sent in order to be burned to transfer to other chains.
-     *
-     * @dev This address is distinct from {BURNED_TX_FEES_ADDRESS} so that the amount of burned transaction
-     * fees and burned transferred amounts can be tracked separately.
-     * This address was chosen arbitrarily.
-     */
-    address public constant BURNED_FOR_TRANSFER_ADDRESS = 0x0100000000000000000000000000000000010203;
-
-    /**
-     * @notice Address used to blackhole funds on the home chain, effectively burning them.
-     *
-     * @dev When reporting burned transaction fee amounts, this address is used as the recipient
-     * address for the funds to be sent to be burned on the home chain.
-     * This address was chosen arbitrarily.
-     */
-    address public constant HOME_CHAIN_BURN_ADDRESS = 0x0100000000000000000000000000000000010203;
-
-    /**
-     * @notice The native minter precompile.
-     */
-    INativeMinter public constant NATIVE_MINTER =
-        INativeMinter(0x0200000000000000000000000000000000000001);
-
-    /**
-     * @dev When modifier is used, the function can only be called after the contract is fully collelateralized,
-     * accounting for the initialReserveImbalance.
-     */
-    modifier onlyWhenCollateralized() {
-        require(getIsCollateralized(), "NativeTokenRemote: contract undercollateralized");
-        _;
-    }
-
-    /**
-     * @notice Initializes this token TokenRemote instance to receive tokens from the specified TokenHome instance,
-     * and represents the received tokens with the native token on this chain.
-     * @param settings Constructor settings for this token TokenRemote instance.
-     * @param nativeAssetSymbol The symbol of the native asset.
-     * @param initialReserveImbalance The initial reserve imbalance that must be collateralized before minting.
-     * @param burnedFeesReportingRewardPercentage_ The percentage of burned transaction fees
-     * that will be rewarded to sender of the report.
-     */
-    function initialize(
+contract NativeTokenRemote is NativeTokenRemoteUpgradeable {
```

Comment:
Can we add a comment explaining that inheriting from the `Upgradeable` version and adding a constructor that calls `initialize` creates a non-upgradeable instance of these contracts?

Reply:
added comments to each non-upgradeable